### PR TITLE
refactor: fetch examples from compiler repo

### PR DESCRIPTION
**Pull request by sockmaster27**
Fetch the drop-down examples from the GitHub repository. This means that it should only be necessary to change the version number to update the examples to accommodate future versions of the language.

Regarding the loading state, the examples are fetched at most a split-second after the rest of the page-content, even with heavy throttling. While present, I find the pop-in effect to look perfectly reasonable.

Fixes flix/flix#5950

### CODE REVIEW

**Comment by sockmaster27** [comment]
IMO I think this is the better solution. We avoid adding the complexity of a network-using build-step, by using a service which is, after all, made for the purpose.

**Comment by magnus-madsen** [comment]
Ah-- but there are two issue still:

- I don't want to depend on GitHub's infra; afterall we cannot know what their rate limits will be in the future.
- We also need more granular control; for example the website is supposed to reflect "the latest stable version", but the master repository might be further ahead, so we still need a mechanism to refer to a specific version. This suggest that having a script (part of the build or not) to generate Samples.js is probably the best way to go.

**Comment by sockmaster27** [comment]
@magnus-madsen 
Regarding the second point, it's already locked to the specific version (v0.38.0), which can of course be changed when updating the website.

**Comment by magnus-madsen** [comment]
Maybe now this can be resolved?

File: package.json [hunk 1]
```diff
@@ -23,9 +23,10 @@
     "reconnecting-websocket": "^4.4.0"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
+    "build-script": "node build-scripts/main.mjs",
+    "start": "npm run build-script & react-scripts start",
```
Comment (magnus-madsen) [inline package.json:27]: Is it right with one ampersand here? In bash that runs the processes concurrently. But maybe not here?

**Comment by magnus-madsen** [comment]
Do we want an empty "Samples.js" (which is overwritten) or perhaps something that indicates "you forgot to run the build-script". Or do we feel good this cannot happen?

I also wonder if we should _not_ run it on the default build. What if you want to do work offline?

**Comment by sockmaster27** [comment]
@magnus-madsen 
I guess it depends on whether you want to have it as just another part of the build-process (like it is now), or its own thing that should be run manually. While the latter is more flexible and more efficient, I think the the way it is now is simpler and less error-prone. IMO failing to load the examples _should_ be an error. 

I'm open to having my mind changed, but I'd also imagine that editing while offline is a pretty rare occurrence in this day and age, and if it should be necessary, you would be able to find the lines to comment out. 

Anyway, I think we may be slightly overthinking this. It was supposed to save time after all :)

**Comment by magnus-madsen** [comment]
Ill merge this and have a look in a few days when I am back.

**Comment by magnus-madsen** [comment]
Thanks!-- seems to work. Just had to update my NPM dependencies.

**Pull request merged**